src/Microsoft.AspNetCore.Mvc.ViewFeatures/IViewComponentHelper.cs

```diff
@@ -16,7 +16,14 @@ public interface IViewComponentHelper
         /// Invokes a view component with the specified <paramref name="name"/>.
         /// </summary>
         /// <param name="name">The name of the view component.</param>
-        /// <param name="arguments">Arguments to be passed to the invoked view component method.</param>
+        /// <param name="arguments">
+        /// Arguments to be passed to the invoked view component method.
+        /// <para>
+        /// The argument can either be an <see cref="System.Collections.Generic.IDictionary{string, object}"/>
+        /// instance, or an anonymous object containing names and values of the arguments for the invoked view
+        /// component method.
```

Comment:
> which the user complained wasn't very useful.

Perhaps w/o the `object` part I could agree. But the framework doesn't care about anonymous `object`s versus any other thing in the system with `public` properties. And the second sentence makes things even more clear.